app/javascript/components/public_health/Import.js

```diff
@@ -79,6 +81,18 @@ class Import extends React.Component {
         });
       })
       .catch(err => {
+        let validation_errors = [];
+        Object.keys(err.response.data).forEach(prop => {
+          validation_errors.push({
+            prop,
+            message: err.response.data[String(prop)],
+          });
+        });
+        const patientIndex = this.state.patients.findIndex(statePatient => _.isEqual(statePatient, patient.patient));
+        let patients = this.state.patients;
+        patient.patient.validation_errors = validation_errors;
+        patients[parseInt(patientIndex)] = patient.patient;
+        this.setState({ patients: patients });
         reportError(err);
```

Comment:
I'd be in favor of making this more descriptive now. At first when you see the red toast pop-up - it looks like Sara Alert failed on the import and didn't tell you anything else. Maybe something like "Records did not pass data validation. See specific errors below." (would need to check with PO team).

Reply:
Are you referencing the 422 error returned?

Reply:
Yep, which is what this function call triggers. We can pass in something more than `err` though, which was my point.